apps/api/src/app/subscribers/subscribers.controller.ts

```diff
@@ -490,16 +509,29 @@ export class SubscribersController {
     @UserSession() user: UserSessionData,
     @Param('subscriberId') subscriberId: string,
     @Body() body: UpdateSubscriberGlobalPreferencesRequestDto
-  ) {
-    const command = UpdateSubscriberGlobalPreferencesCommand.create({
-      organizationId: user.organizationId,
-      subscriberId,
-      environmentId: user.environmentId,
-      enabled: body.enabled,
-      preferences: body.preferences,
-    });
+  ): Promise<Omit<UpdateSubscriberPreferenceResponseDto, 'template'>> {
+    const channels = body.preferences?.reduce((acc, curr) => {
+      acc[curr.type] = curr.enabled;
```

Comment:
Is this enabled flag the same as the one we talked about in skipped tests?

Reply:
No, this is the channel-level flag. There are 2 `enabled` flags:
* `enabled` at top level of V1 Preferences, which controls whether to use the subscriber preference during preference merging
* `enabled` flag for each channel, which controls delivery for the channel